packages/navigation/src/hooks/useCurrentUrl.ts

```diff
@@ -1,16 +1,44 @@
-import { useNavigation } from '@react-navigation/native';
+import { LinkingOptions, useNavigation } from '@react-navigation/native';
 
-export function useCurrentUrl(defaultBaseUrl?: string) {
+export type LinkingConfig = LinkingOptions<{}>['config'];
+
+function findPath(
+  name: string | undefined,
+  config: LinkingConfig
+): string | undefined {
+  if (!config || !name) return undefined;
+  const screens = config.screens;
+  for (const key of Object.keys(screens)) {
+    // @ts-expect-error
+    const pathOrScreen: string | LinkingConfig = screens[key];
+    if (typeof pathOrScreen === 'string') {
+      if (key === name) {
+        return pathOrScreen;
+      }
+    } else {
+      const path = findPath(name, pathOrScreen);
+      if (path) {
+        return path;
+      }
+    }
+  }
+  return undefined;
+}
+
+function getPath(params: unknown) {
+  if (params && typeof params === 'object' && 'path' in params) {
+    return params.path;
+  }
+  return undefined;
+}
+
+export function useCurrentUrl(baseUrl: string, config: LinkingConfig) {
```

Comment:
We must take params into consideration here, to support the use case where we navigate to a fallback route with params, eg: `/one_page?some_param=true`

In this case, the `getCurrentUrl()` should include the params as well and not just the path.